security/grub-extend-pcr.py

```diff
@@ -0,0 +1,102 @@
+        if 'POWER10' not in val:
+            self.cancel("Power10 LPAR is required to run this test.")
+        device_tree_path = "/proc/device-tree/vdevice/"
+        d_list = os.listdir(device_tree_path)
+        vtpm = [i for i, item in enumerate(d_list) if item.startswith('vtpm@')]
+        if not vtpm:
+            self.cancel("vTPM not enabled.")
+        smm = SoftwareManager()
+        deps = []
+        detected_distro = distro.detect()
+        if detected_distro.name in ['rhel', 'fedora', 'centos', 'redhat']:
+            deps.extend(["tss2"])
+        elif 'SuSE' in detected_distro.name:
+            deps.extend(['ibmtss'])
+        else:
+            self.cancel("%s not supported for this test" %
+                        detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+    def _run_pcrread(self, cmd):
+        '''
+        Run tsspcrread command, validate output
+        '''
+        output = process.system_output(cmd, ignore_status=True).decode().splitlines()
+        if '00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00' in output:
+            self.fail("%s Failed to give correct output." % cmd)
+        '''
+        Fail case:
+        count 1 pcrUpdateCounter 39
+        digest length 32
+        00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+        00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+        '''
+
+    def test_tsspcrread_8(self):
+        self._run_pcrread("tsspcrread -ha 8")
+
+    def test_tsspcrread_9(self):
+        self._run_pcrread("tsspcrread -ha 9")
+
+    def test_tsseventextend(self):
+        '''
+        Run tsseventextend command, validate output
+        '''
+        cmd = "tsseventextend -if /sys/kernel/security/tpm0/binary_bios_measurements -sim -pcrmax 9"
+        output = process.system_output(cmd, ignore_status=True).decode().splitlines()
+        pcr8_value = ""
+        pcr9_value = ""
+        pcr8_flag = True
+        pcr9_flag = True
+        for line in output:
+            if 'PCR 08' in line:
+                pcr8_value = line.split(":")[1]
+            if 'PCR 09' in line:
+                pcr9_value = line.split(":")[1]
+        if "00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00" in pcr8_value:
+            pcr8_flag = False
+        if "00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00" in pcr9_value:
+            pcr9_flag = False
+        if not (pcr8_flag and pcr9_flag):
+            self.fail("PCR 8 and 9 not having correct values.")
+        '''
+        Fail case:
+        PCR 08: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+        PCR 09: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+        '''
```

Comment:
the above comments will look clean in method doc string or readme

Reply:
Resolved.